File: Check If All 1's Are at Least Length K Places Away.c


bool kLengthApart(int* nums, int numsSize, int k){
    int i, count = 0;
    bool flag = false;
    for(i = 0 ; i < numsSize; i++)
    {
        if (nums[i] == 1)
        {
            if(count < k && flag)
            {
                return false;
            }
            flag = true;
            count = 0;
        }
        if(flag && nums[i] == 0 )
        {
            count += 1;
        }
    }
    return true;
}

File: Check If All 1's Are at Least Length K Places Away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        count = 0
        flag =False
        for i in nums:
            if i==1:
                if count<k and flag ==True:
                    return False
                flag = True
                count = 0
            if flag == True and i==0:
                count+=1
        return True
